server/sms/sms_gateway_test.go

```diff
@@ -231,3 +233,104 @@ func TestReadLastID(t *testing.T) {
 
 	a.Equal(uint64(10), gw.LastIDSent)
 }
+
+func tempFilename() string {
+	file, err := ioutil.TempFile("/tmp", "guble_sms_retry")
+	if err != nil {
+		panic(err)
+	}
+	file.Close()
+	os.Remove(file.Name())
+	return file.Name()
+}
+
+func Test_RetryLoop(t *testing.T) {
+	ctrl, finish := testutil.NewMockCtrl(t)
+	defer finish()
+
+	defer testutil.EnableDebugForMethod()()
+	a := assert.New(t)
+
+	//create a mockSms Sender to simulate Nexmo error
+	mockSmsSender := NewMockSender(ctrl)
+	f := tempFilename()
+	defer os.Remove(f)
+
+	//create a KVStore with sqlite3
+	kvStore := kvstore.NewSqliteKVStore(f, false)
+	err := kvStore.Open()
+	a.NoError(err)
+	a.NotNil(kvStore)
+
+	//create a new mock router
+	routerMock := NewMockRouter(testutil.MockCtrl)
+	routerMock.EXPECT().KVStore().AnyTimes().Return(kvStore, nil)
+
+	//create a new mock msg store
+	mockMessageStore := NewMockMessageStore(ctrl)
+	routerMock.EXPECT().MessageStore().AnyTimes().Return(mockMessageStore, nil)
+	mockMessageStore.EXPECT().MaxMessageID(gomock.Any()).Return(uint64(2), nil)
+
+	//setup a new sms gateway
+	worker := 8
```

Comment:
what happens if other workers (not the one retrying to send) are sending correct messages at the same time? 
is the lastMessageID correctly updated ?